### PR TITLE
fix: prevent slide upload by participants after session (#10582)

### PR DESCRIPTION
**Pull request by jimfenton**
Restricts slide submission by other than those that can manage slides after the session has ended.

Does not include tests for this, and breaks two existing tests:

======================================================================
FAIL: test_propose_session_slides (ietf.meeting.tests_views.MaterialsTests.test_propose_session_slides)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/ietf/meeting/tests_views.py", line 7077, in test_propose_session_slides
    self.assertTrue(q('.proposeslides'))
AssertionError: [] is not true

======================================================================
FAIL: test_submit_and_approve_multiple_versions (ietf.meeting.tests_views.MaterialsTests.test_submit_and_approve_multiple_versions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dev/.local/lib/python3.12/site-packages/django/test/utils.py", line 461, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/unittest/mock.py", line 1396, in patched
    return func(*newargs, **newkeywargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/ietf/meeting/tests_views.py", line 7271, in test_submit_and_approve_multiple_versions
    self.assertEqual(r.status_code, 302)
AssertionError: 403 != 302

Apparently this was because the sessions created for testing did not specify a time. I'm surprised this didn't fail before, because the submission would probably have violated the grace time after which Secretariat action is needed. I tried but didn't manage to succeed in fixing these test cases by adding a session time to the SessionFactory calls.

### CODE REVIEW

**Comment by jimfenton** [comment]
I found another "Upload slides" button that isn't covered by this fix yet (materials/views.py line 1937). Will add.

File: ietf/meeting/views.py [hunk 1]
```diff
@@ -1931,7 +1931,9 @@ def api_get_session_materials(request, session_id=None):
 
     minutes = session.minutes()
     slides_actions = []
-    if can_manage_session_materials(request.user, session.group, session) or not session.is_material_submission_cutoff():
+    if has_role(request.user, "Secretariat") \
+            or (not session.is_material_submission_cutoff() and session.can_manage_materials(request.user)) \
+            or not session.is_past():
         slides_actions.append(
```
Comment (jennifer-richards) [inline ietf/meeting/views.py]:
```suggestion
    if (
        has_role(request.user, "Secretariat")
        or (not session.is_material_submission_cutoff() and session.can_manage_materials(request.user))
        or not session.is_past()
    ):
        slides_actions.append(
```

style nit

File: ietf/templates/meeting/session_details_panel.html [hunk 2]
```diff
@@ -185,7 +185,7 @@ <h3 class="mt-4">Slides</h3>
                href="{% url 'ietf.meeting.views.upload_session_slides' session_id=session.pk num=session.meeting.number %}">
                 Upload new slides
             </a>
-        {% elif request.user.is_authenticated and not session.is_material_submission_cutoff %}
+        {% elif request.user.is_authenticated and not session.is_past %}
```
Comment (jennifer-richards) [inline ietf/templates/meeting/session_details_panel.html:188]: Should this be `...and not session.is_past and not session.is_material_submission_cutoff`? I.e., is this going to undesirably start allowing submissions between the material submission cutoff and the end of the session?

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
Style nit + a question

File: ietf/templates/meeting/session_details_panel.html (hunk 2 above)
Comment (jimfenton) [reply ietf/templates/meeting/session_details_panel.html:188]: It was my understanding that the material submission cutoff was after the meeting, to lock down the content for proceedings and such. This was partly based on what I saw in the test cases. So is_past would be a tighter restriction. If my understanding was incorrect, this would need to change.

Comment (jennifer-richards) [reply ietf/templates/meeting/session_details_panel.html:188]: Ah, you're likely right - I was thinking of draft submission deadline.

**Review by jennifer-richards — APPROVED** [review]
Fixed the failing tests, added missing test coverage, and adjusted to handle a corner case.
